src/ImageSharp/Formats/ImageFormatBase{T}.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+
+namespace SixLabors.ImageSharp.Formats
+{
+    /// <summary>
+    /// The base class for all image formats.
+    /// Inheriting classes should implement the singleton pattern by creating a private constructor.
+    /// </summary>
+    /// <typeparam name="T">The type of image format.</typeparam>
+    public abstract class ImageFormatBase<T> : IImageFormat
```

Comment:
The drawback of this class is that it pollutes the public API by complicating the class hierarchy with implementation details, with no benefits IMO. The reason is that there is no need for lazy instance creation, because the constructor is trivial and brings no thread safety issues.

Defining a format without it is actually simpler:
```C#
    public sealed class GifFormat : IImageFormat
    {
        private GifFormat() { }

        public string Name => "GIF";

        public string DefaultMimeType => "image/gif";

        public IEnumerable<string> MimeTypes => GifConstants.MimeTypes;
        
        public IEnumerable<string> FileExtensions => GifConstants.FileExtensions;
        
        // The only extra line:
        public static GifFormat Instance { get; } = new GifFormat();
    }
```

Reply:
Yeah, what I wanted though was a means of enforcing the standard. I'm still not sure.